### PR TITLE
Indexer: use $key:$version as an id for ES docs

### PR DESCRIPTION
**Pull request by nl0**

### CODE REVIEW

File: lambdas/es/indexer/document_queue.py [hunk 1]
```diff
@@ -124,6 +130,8 @@ def append(
             # TODO remove this; it's not meaningful since we use a different index
             # type for object vs. package documents
             "_type": "_doc",
+            # Elastic native keys
```
Comment (sir-sigurd) [inline lambdas/es/indexer/document_queue.py:133]: I'm not sure if this comment adds any value...

File: lambdas/es/indexer/document_queue.py [hunk 2]
```diff
@@ -43,6 +43,12 @@
 RETRY_429 = 3
 
 
+def get_id(key, version_id):
+    """guarantee primary key uniqueness for package delete (marker) documents"""
+    # TODO: use smth like _hash(key) + _hash(version_id) for ids to overcome field length resctriction (512 bytes)
```
Comment (sir-sigurd) [inline lambdas/es/indexer/document_queue.py:48]: Might make sense to add a reference for the length restriction: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html.

File: lambdas/es/indexer/document_queue.py [hunk 3]
```diff
@@ -43,6 +43,12 @@
 RETRY_429 = 3
 
 
+def get_id(key, version_id):
+    """guarantee primary key uniqueness for package delete (marker) documents"""
```
Comment (sir-sigurd) [inline lambdas/es/indexer/document_queue.py:47]: The comment seems a bit misleading since it guarantees uniqueness for all S3 objects.

**Review by sir-sigurd — APPROVED** [review]
There are some nitpicks, otherwise LGTM.

File: lambdas/es/indexer/document_queue.py (hunk 2 above)
Comment (nl0) [reply lambdas/es/indexer/document_queue.py:48]:
```suggestion
    # TODO: use smth like _hash(key) + _hash(version_id) for ids to overcome [512 bytes length restriction](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html)
```

Comment (sir-sigurd) [inline lambdas/es/indexer/document_queue.py:48]:
```suggestion
    """
    Generate unique value for every object in the bucket to be used as 
    document `_id`. This value must not exceed 512 bytes in size:
    https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html.
    # TODO: both object key and version ID are up to 1024 bytes long, so 
    # we need to use something like `_hash(key) + _hash(version_id)` to 
    # overcome the mentioned size restriction.
    """
```

File: lambdas/es/indexer/document_queue.py [hunk 4]
```diff
@@ -43,6 +43,12 @@
...
+    # TODO: use smth like _hash(key) + _hash(version_id) for ids to overcome [512 bytes length restriction](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html)
```
Comment (sir-sigurd) [inline lambdas/es/indexer/document_queue.py:48]:
```suggestion
    """
    Generate unique value for every object in the bucket to be used as 
    document `_id`. This value must not exceed 512 bytes in size:
    https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html.
    # TODO: both object key and version ID are up to 1024 bytes long, so 
    # we need to use something like `_hash(key) + _hash(version_id)` to 
    # overcome the mentioned size restriction.
    """
```

**Pull request merged**